docs/allocation.md

```diff
@@ -0,0 +1,152 @@
+is increased by an increment (default 4 KB). The requested size can grow up to a predefined maximum (default 128 KB).
+
+After every garbage collection cycle, the TLH requested size for each thread is reduced, sometimes by as much as 50%, to take account of threads that
+reduce their allocation rate and no longer need large TLHs.
+
+For very inactive threads, the requested size can even drop below the initial value, down to the predefined minimum (512/768 bytes).
+For very active threads, the maximum TLH requested size might be reached before the next GC occurs.
+
+Larger TLHs can help reduce heap lock contention, but might also reduce heap utilization and increase heap fragmentation.
+
+The following options control the requested TLH size:
+
+- [`-Xgc:tlhMaximumSize=<bytes>`](xgc.md#tlhmaximumsize)
+- [`-Xgc:tlhInitialSize=<bytes>`](xgc.md#tlhinitialsize)
+- [`-Xgc:tlhIncrementSize=<bytes>`](xgc.md#tlhincrementsize)
+
+Typically, when the maximum TLH size is increased, you should also increase the increment proportionally, so that active threads can
+reach the maximum requested TLH size more quickly.
+
+## Heap configuration
+
+Depending on the memory management strategy in force, the Java heap can be configured in a number of ways. The simplest configuration consists of a single area of memory, often referred to as a *flat* heap. Other configurations divide the heap into different *areas* or *regions*, which might contain objects of different ages (generations) or sizes.
+
+### Area-based heaps
+
+The default GC policy for OpenJ9 uses a heap configuration that is divided into two main areas: the *nursery* area for new object allocation, and the *tenure* area for objects that continue to be reachable for a longer period of time. Most objects have short lifecycles and can be reclaimed by the garbage collector more quickly by focusing only on the nursery area. Global GC cycles that cause application pauses in order to clear and defragment the *tenure* area are less frequent.
+
+### SOA and LOA
+
+Some GC policies subdivide a *flat* heap or an *area* of the heap for object allocation into the Small Object Area (SOA) and the Large Object Area (LOA).
+
+The allocator initially attempts to allocate objects in the SOA, regardless of size. If the allocation cannot be satisfied the following actions are possible, depending on object size:
+
+- If the object is smaller than 64 KB, an allocation failure occurs, which triggers a GC action.
+- If the object is larger than 64 KB, the allocator attempts to allocate the object in the LOA. If the allocation cannot be satisfied, an allocation failure occurs, which triggers a GC action.
+
+The GC action that is triggered by the allocation failure depends on the GC policy in force.
+
+The overall size of the LOA is calculated when the heap is initialized, and recalculated at the end of each global GC cycle. The GC can expand or shrink the LOA, depending on usage, to avoid allocation failures.
```

Comment:
We far more often use (in conversation, in code, VGC....) term 'contract' than 'shrink' (not just for LOA, but any type of heap resizing). There are VGC stanzas that report heap resizing elements, and the possible types are expand and contract. Consider aligning the documentation with it.

Reply:
All instances of updated to use the term *contract* or *contraction*.